arm/compute/client.go

```diff
@@ -0,0 +1,2031 @@
+	}
+
+	req, err = autorest.Prepare(
+		req,
+		client.WithAuthorization(),
+		client.WithInspection())
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure sending VirtualMachineExtensionImagesClient.Get request (%v)", err)
+	}
+
+	resp, err := autorest.SendWithSender(client, req)
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure sending VirtualMachineExtensionImagesClient.Get request (%v)", err)
+	}
+
+	result = VirtualMachineExtensionImage{}
+	err = autorest.Respond(
+		resp,
+		client.ByInspecting(),
+		autorest.WithErrorUnlessOK(),
+		autorest.ByUnmarshallingJSON(&result),
+		autorest.ByClosing())
+
+	return result, err
+}
+
+func (client *VirtualMachineExtensionImagesClient) NewGetRequest(location string, publisherName string, type__ string, version string) (*http.Request, error) {
+
+	pathParameters := map[string]interface{}{
+		"location":       url.QueryEscape(location),
+		"publisherName":  url.QueryEscape(publisherName),
+		"subscriptionId": url.QueryEscape(client.SubscriptionId),
+		"type":           url.QueryEscape(type__),
+		"version":        url.QueryEscape(version),
+	}
+
+	queryParameters := map[string]interface{}{
+		"api-version": ApiVersion,
+	}
+
+	return autorest.DecoratePreparer(
+		client.GetRequestPreparer(),
+		autorest.WithPathParameters(pathParameters),
+		autorest.WithQueryParameters(queryParameters)).Prepare(&http.Request{})
+}
+
+func (client *VirtualMachineExtensionImagesClient) GetRequestPreparer() autorest.Preparer {
+	return autorest.CreatePreparer(
+		autorest.AsJSON(),
+		autorest.AsGet(),
+		autorest.WithBaseURL(client.BaseUri),
+		autorest.WithPath("/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/publishers/{publisherName}/artifacttypes/vmextension/types/{type}/versions/{version}"))
+}
+
+// ListVersions gets a list of virtual machine extension image versions.;
+//
+// filter is the filter to apply on the operation.
+func (client *VirtualMachineExtensionImagesClient) ListVersions(location string, publisherName string, type__ string, filter string, top int, orderby string) (result []VirtualMachineImageResource, err error) {
+
+	req, err := client.NewListVersionsRequest(location, publisherName, type__, filter, top, orderby)
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure preparing VirtualMachineExtensionImagesClient.ListVersions request (%v)", err)
+	}
+
+	req, err = autorest.Prepare(
+		req,
+		client.WithAuthorization(),
+		client.WithInspection())
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure sending VirtualMachineExtensionImagesClient.ListVersions request (%v)", err)
+	}
+
+	resp, err := autorest.SendWithSender(client, req)
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure sending VirtualMachineExtensionImagesClient.ListVersions request (%v)", err)
+	}
+
+	result = []VirtualMachineImageResource{}
```

Comment:
@ahmetalpbalkan And they're the same size. See http://play.golang.org/p/Rho7YDbGDE.

Leaving it as it is leaves the generator less complicated (otherwise we have to add a case to handle this...).